include/pybind11/stl_bind.h

```diff
@@ -636,18 +639,58 @@ auto map_if_insertion_operator(Class_ &cl, std::string const &name)
         "Return the canonical string representation of this map.");
 }
 
-template <typename Map>
+template <typename KeyType>
 struct keys_view {
-    Map &map;
+    virtual size_t __len__() = 0;
```

Comment:
Huh, I think this is the first non-optin virtual polymorphism we are using in pybind11, I wonder if this will affect any of our downstream users. I suppose this is not that big of an issue since it's in an optional include though. Thoughts?